Signed-off-by: rasool-thesis <rasool.moradi@students.uniroma2.eu>

rasool-thesis opened the pull request:
first step just change the paths on the file with a static command

Commit:
trying-automation.text

Commit:
Update logic in CombinedExperiment thread

Commit:
This commit replaces the legacy Bash-based experiment automation with a structured and maintainable Python implementation:
•	Reimplements the experiment runner logic in Python (experiment_runner.py)
•	Supports dynamic kernel switching and remote rebooting
•	Executes and monitors TRex experiments using TrexDriverCLI4_gpt.py
•	Logs CPU load and traffic data in a structured format
•	Automatically builds test output filenames based on config
•	Simplifies future extensions and config reuse
This update improves readability, cross-platform compatibility, and integration with larger Python-based systems.

Signed-off-by: Rmoradi <Rmoradi@tg.firsttest1.superfluidity-pg0.wisc.cloudlab.us>

Commit:
Add centralized configuration file for experiment control

- Extracts kernel list, test parameters, and paths into a separate config module
- Allows users to modify testbed settings without touching the main logic
- Supports flexibility for changing duration, packet rate, output directories

Signed-off-by: Rmoradi <Rmoradi@tg.firsttest1.superfluidity-pg0.wisc.cloudlab.us>

Commit:
Rewrite experiment...sh from bash to modular Python

- Converts nested kernel-based experiment control to Python
- Supports dynamic experiment configurations through a config file
- Handles kernel switching, rebooting, and SUT initialization via SSH
- Clean and readable JSON output capturing TX/RX and CPU stats
- Strong structure for reproducibility and multi-kernel testing

Signed-off-by: Rmoradi <Rmoradi@tg.firsttest1.superfluidity-pg0.wisc.cloudlab.us>

rasool-thesis committed:
Delete scripts/TrexDriverCLI4_gpt.py

no more need

rasool-thesis committed:
Delete scripts/TrexDriverCLI4_gpt_onethread.py

update with some changes

rasool-thesis committed:
Add files via upload

Merge dual-threaded TREX + netrace control into a unified execution thread

- Replaces separate threads with a single CombinedExperiment thread
- Executes TREX locally and netrace remotely in a coordinated flow
- Fetches structured CPU load data via SSH from the remote server
- Parses final CPU results from remote output as JSON
- Outputs total TX/RX and CPU metrics clearly with random ID tagging
- Ensures the code is modular and SSH-secure using Paramiko

rasool-thesis committed:
Delete scripts/exper-exper.py

no more needed

rasool-thesis committed:
Delete scripts/experiment-gpt.sh

no more needed

rasool-thesis committed:
Delete scripts/experiment-run.py

no more needed

rasool-thesis committed:
Add files via upload

Refactor cpu_load_netrace_gpt.py to use JSON and dictionary-based logging

- Parses structured CPU usage data from JSON-formatted netrace output
- Computes mean and standard deviation for EVENT_NET_RX_SOFTIRQ
- Organizes output using a dictionary for cleaner downstream usage
- Adds error handling for malformed JSON or missing fields
- Saves output in a dedicated directory relative to the script location

rasool-thesis committed:
Add files via upload

feat(netrace): convert netrace runner from Bash to Python with JSON config support

- Replaces legacy start_netrace.sh Bash script with structured Python logic
- Reads duration and experiment ID from external JSON configuration
- Creates output directory automatically and saves logs with consistent naming
- Starts netrace using subprocess, handles duration and graceful SIGINT
- Improves portability, maintainability, and ease of customization

rasool-thesis committed:
Update experiment-run-withconfig.py

update

plungaroni commented on scripts/TrexDriverCLI4_gpt_onethread.py:
```diff
@@ -0,0 +1,133 @@
+        private_key = paramiko.RSAKey.from_private_key_file(self.pkey_path)
+        ssh.connect(self.remote, port=self.port, username=self.user, pkey=private_key)
+
+        script_dir = "/users/Rmoradi/pastrami/scripts"
```
Did you hardcode the path?

rasool-thesis replied:
ohh I did not see it, ok I changed it in updates.

plungaroni commented on scripts/TrexDriverCLI4_gpt_onethread.py:
```diff
@@ -0,0 +1,133 @@
+
+        script_dir = "/users/Rmoradi/pastrami/scripts"
+        command = f"cd {script_dir} && python3 cpu_load_netrace_gpt.py {self.cpu_id} {self.duration} {self.rnd_id} dummy_path"
+        stdin, stdout, stderr = ssh.exec_command(command)
```
Have you checked that this command is not blocking?

rasool-thesis replied:
yes absolutely, you are right if it goes more than 30 seconds there will be problem. I solved it in new update with its commit.

plungaroni commented on scripts/TrexDriverCLI4_gpt_onethread.py:
```diff
@@ -0,0 +1,133 @@
+        driver = TrexDriver(self.server, self.txPort, self.rxPort, self.pcap, self.rate, self.duration)
+        self.output_local = driver.run()
+
+        self.output_remote = stdout.read().decode() + stderr.read().decode()
```
If the code on line 40 is not blocking, how are you sure the data is available?

rasool-thesis replied:
solved

plungaroni commented on scripts/TrexDriverCLI4_gpt_onethread.py:
```diff
@@ -0,0 +1,133 @@
+
+        print("cpu_load_output:")
+
+        last_brace_open = output_remote.rfind('{')
```
Why do you do this?
if `output_remote` is json isn't there a simpler way to check if it exists and parse it?

plungaroni commented on scripts/TrexDriverCLI4_gpt_onethread.py:
```diff
@@ -0,0 +1,133 @@
+
+        args = parser.parse_args()
+
+        combined_thread = CombinedExperiment(
```
The reason I made two threads was because both `cpu_netrace` and `Trex` were blocking.

If it is still like that I have the impression that, as the code is written, first it executes `cpu_netrace` (and the program blocks) finished netrace starts `Trex`.

So, if I am right, the two measurements are sequential and not parallel.

rasool-thesis replied:
I changed it in update now they work in parallel.

plungaroni commented on scripts/config.py:
```diff
@@ -0,0 +1,29 @@
+import os
+import datetime
+
+# Experiment Configuration
```
Ok, to externalize the configuration, but shouldn't it have been done with a YAML or JSON file?

Why did he do this Python file?
If you wrote it in python, should you have put it directly in `experiment-run-withconfig.py` ?

rasool-thesis replied:
certainly you are right. I made changes

rasool-thesis replied:
I changed it to the yaml and update it

plungaroni commented on scripts/start_netrace.py:
```diff
@@ -0,0 +1,45 @@
+output_dir = os.path.join(script_dir, 'rm')
+
+# Make sure the 'rm' directory exists (create if not)
+os.makedirs(output_dir, exist_ok=True)
```
Why do you put everything in this rm folder inside the folder pointed to by `script_dir` ?

Isn't it better to make a separate folder, maybe outside the `script_dir` path?

rasool-thesis replied:
it was for checking I will update it

rasool-thesis replied:
change to root has been done

plungaroni commented on scripts/start_netrace.py:
```diff
@@ -0,0 +1,45 @@
+with open(output_filename, 'w') as outfile:
+    print("Starting /root/netrace...")
+    process = subprocess.Popen(
+        ['/root/netrace'],
```
Here too the `netrace` program path is absolute.

It should be relative.
And we could assume the conversion that the `netrace` program is also in the scripts folder.

rasool-thesis replied:
yes you are right change has been done

plungaroni commented on scripts/start_netrace.py:
```diff
@@ -0,0 +1,45 @@
+    print(f"%%% PID: {process.pid} %%%")
+
+    # Run for the specified duration
+    time.sleep(duration)
```
This is a blocking statement (waits for "`duration`" seconds).

And it is at the end of the call chain to start netrace.
(_TrexDriverCLI4_gpt_onethread.py_ -> _cpu_load_netrace_gpt.py_ -> _start_netrace.py_)

So also the command on line 40 in _TrexDriverCLI4_gpt_onethread.py_ (`stdin, stdout, stderr = ssh.exec_command(command)`) stops waiting for netrace scripts to finish.
Since it is called in the same thread as Trex I suspect that these two are called sequentially and not in parallel as desired.

Please check.

rasool-thesis replied:
I have updated the design and now ssh.exec_command that triggers start_netrace.py and the local TrexDriver.run() are executed in two separate threads.

The remote netrace starts inside its own thread (RemoteNetrace).

The local Trex starts in parallel in another thread (LocalTrex).

Therefore, even though start_netrace.py includes a blocking sleep(duration), it doesn't block the local T-Rex anymore, because the SSH session runs in a separate thread.
 Now netrace and T-Rex run truly in parallel, not sequentially.

plungaroni commented on scripts/start_netrace.py:
```diff
@@ -0,0 +1,45 @@
+import signal
+import os
+
+# Load config from JSON
```
Why pass two parameters via file.

Wasn't it easier to do it from the command line as previously done?

rasool-thesis replied:
as this is the start point of dictionary it is easier to create a file and from now on I go through with it, and if someone else come and run our scripts it can be easier for them to understand and make troubleshooting if some were made error.

plungaroni commented on scripts/cpu_load_netrace_gpt.py:
```diff
@@ -0,0 +1,107 @@
+    cpu_id = int(sys.argv[1])
+    duration = int(sys.argv[2])
+    exp_id = sys.argv[3]
+    output_path = sys.argv[4]  # Not used, placeholder for compatibility
```
If you don't use it, why keep it?

You wrote the main files from scratch that call each other so there are no compatibility issues.

rasool-thesis replied:
just this line did not used    output_path = sys.argv[4]  # Not used, placeholder for compatibility
so I delete it in new codes

plungaroni commented on scripts/cpu_load_netrace_gpt.py:
```diff
@@ -0,0 +1,107 @@
+    }
+
+    script_dir = os.path.dirname(os.path.abspath(__file__))
+    config_path = os.path.join(script_dir, 'rm_config.json')
```
Apparently you pass parameters to start_netrace.py via file.

Why? (see comment in _start_netrace.py_)

Then pay attention to the file name. Here you save a rm_config.json, while in _start_netrace.py_ you read config.json

Please check.

rasool-thesis replied:
yes you are right it was a writing mismatch, I done it

plungaroni commented on scripts/start_netrace.py:
```diff
@@ -0,0 +1,45 @@
+import os
+
+# Load config from JSON
+with open('config.json') as f:
```
Are you sure the file name is correct?

rasool-thesis replied:
changes has been made

plungaroni commented on scripts/cpu_load_netrace_gpt.py:
```diff
@@ -0,0 +1,107 @@
+
+    # Run start_netrace.py with config
+    start_netrace_path = os.path.join(script_dir, 'start_netrace.py')
+    subprocess.run([sys.executable, start_netrace_path, config_path], check=True)
```
You pass `config_path` as a command line parameter, but then in _start_netrace.py_ you don't read it.

Why?

rasool-thesis replied:
yes thanks for your review I made a wrong thing but now I changed it as it be set with command line such :
import sys

if len(sys.argv) != 2:
    print("Usage: python start_netrace.py <config_path>")
    sys.exit(1)

config_path = sys.argv[1]

with open(config_path) as f:
    config = json.load(f)
now it work well

plungaroni commented on scripts/cpu_load_netrace_gpt.py:
```diff
@@ -0,0 +1,107 @@
+        "duration_sec": duration,
+        "mean_cpu_load": round(mean_cpu_load, 2),
+        "std_dev_cpu_load": round(std_dev_cpu_load, 2),
+        "samples_analyzed": len(selected_cpu_loads)
```
You added this additional metric. Good.

But then in _TrexDriverCLI4_gpt_onethread.py_ you don't read it.
Why?

rasool-thesis replied:
surely it was again a mistake of my files I forgot it to upload the correct file, now it is fixed with the command:print(f"samples_analyzed: {cpu_result['samples_analyzed']}")
and the result will be such as screen shot
![new](https://github.com/user-attachments/assets/85de22f3-5f4b-45dd-a145-7bba04a06786)

plungaroni commented on scripts/config.py:
```diff
@@ -0,0 +1,29 @@
+KERNELS = ["k5.15"]
+#KERNEL= ["k5.10" "k5.12" "k5.15" "k5.19" "k6.2" "k6.5" "k6.8" "k6.10" "k6.12" "k6.13" "k6.14"]
+# Paths
+BASE_DIR = os.getenv("PASTRAMI_BASE_DIR", os.path.dirname(os.path.abspath(__file__)))
```
Check the paths generated by lines 22, 23, 24.

Something seems strange to me.

rasool-thesis replied:
change done every things right now

plungaroni commented on scripts/config.py:
```diff
@@ -0,0 +1,29 @@
+os.makedirs(DIR, exist_ok=True)
+
+# Other
+PCAP_PATH = "/proj/superfluidity-PG0/srperf2/pcap/trex-pcap-files/plain-ipv6-64.pcap"
```
It would also be better if this file is in a relative path and internal to the repository.

Perhaps in a folder dedicated to pcap.

rasool-thesis replied:
surly It is a very important point I have made this pastrami/
├── scripts/
│   └── experiment-run.py
├── pcap/
│   └── plain-ipv6-64.pcap
├── config.yml
so from now it is a part of portable folder pastrami

plungaroni commented on scripts/experiment-run-withconfig.py:
```diff
@@ -0,0 +1,95 @@
+        print(f"Switching kernel to version {krl[1:]}")
+        krl_file = subprocess.check_output(
+            ["ssh", f"root@{IP_REMOTE}", f"ls /boot/vmlinuz* | grep -F {krl[1:]}"], text=True).strip()
+        subprocess.run(["ssh", f"root@{IP_REMOTE}", f"{SCRIPTS_DIR}/boot-kernel.sh {krl_file}"])
```
Why do it this weird way ???

You can use the paramiko module to send ssh commands in python as already done in TrexDriverCLI4_gpt_onethread.py

plungaroni commented on scripts/experiment-run-withconfig.py:
```diff
@@ -0,0 +1,95 @@
+        subprocess.run(["ssh", f"root@{IP_REMOTE}", f"{SCRIPTS_DIR}/boot-kernel.sh {krl_file}"])
+        subprocess.run(["ssh", f"root@{IP_REMOTE}", "reboot"])
+        print("Waiting 5 minutes for reboot...")
+        subprocess.run(["sleep", "300"])
```
This seems stupid to me.

Why run the linux program "_sleep_" via bash, **when python can do it natively**???

`time.sleep(300)`

and yet you have already used it, for example, in _start_netrace.py_ line 38.

plungaroni commented on scripts/experiment-run-withconfig.py:
```diff
@@ -0,0 +1,95 @@
+        subprocess.run(["ssh", f"root@{IP_REMOTE}", f"{SCRIPTS_DIR}/sut_init.sh"])
+        subprocess.run(["ssh", f"root@{IP_REMOTE}", "uname -a"])
+        print("Waiting 30 seconds before starting...")
+        subprocess.run(["sleep", "30"])
```
Same stupid thing as line 81.

So same comment.

plungaroni commented on scripts/experiment-run-withconfig.py:
```diff
@@ -0,0 +1,95 @@
+        subprocess.run(["ssh", f"root@{IP_REMOTE}", "reboot"])
+        print("Waiting 5 minutes for reboot...")
+        subprocess.run(["sleep", "300"])
+        subprocess.run(["ssh", f"root@{IP_REMOTE}", f"{SCRIPTS_DIR}/sut_init.sh"])
```
Also here for line 82 and 83.

There is paramiko to use ssh in python why not use it?

plungaroni commented on scripts/experiment-run-withconfig.py:
```diff
@@ -0,0 +1,95 @@
+            "python3", os.path.join(SCRIPTS_DIR, "TrexDriverCLI4_gpt.py"),
+            "-s", "127.0.0.1",
+            "-r", IP_REMOTE,
+            "-c", "4",
```
The `-c` parameter represents the _CPU_ID_ of the cpu to monitor.

In our implementation we chose to use cpu number 4 for our experiments in cloudlab, but this is not true in general.

It would be good if this parameter was editable and not hardcoded.

plungaroni commented on scripts/experiment-run-withconfig.py:
```diff
@@ -0,0 +1,95 @@
+            "-s", "127.0.0.1",
+            "-r", IP_REMOTE,
+            "-c", "4",
+            "-o", "22",
```
same thing for parameters `-o`, `-u`, `--txPort`, `--txPort`
(lines 17, 18, 19, 20)

plungaroni commented on scripts/experiment-run-withconfig.py:
```diff
@@ -0,0 +1,95 @@
+            print(result)
+
+            pid = extract_value(result, "random_id:")
+            duration_sec = extract_value(result, "duration_sec:")
```
I don't see _TrexDriverCLI4_gpt_onethread.py_ leaving `duration_sec:` as a result.

Can you check, please?

plungaroni commented on scripts/experiment-run-withconfig.py:
```diff
@@ -0,0 +1,95 @@
+            result = subprocess.check_output(cmd, text=True)
+            print(result)
+
+            pid = extract_value(result, "random_id:")
```
What is this `pid`?

And why do you associate it with `random_id`?

plungaroni commented on scripts/experiment-run-withconfig.py:
```diff
@@ -0,0 +1,95 @@
+            exp_record = [
+                [rate, 4],
+                {
+                    "PID": pid,
```
I think you are getting a bit confused here.

What do we do with this pid, which is a random id?

Don't we have to take the values ​​of transmitted and received packets from Trex?

(txPackets, rxPackets)

Please check this part carefully and, if it can help you, also compare with my original program in bash.

**Here I don't see how the two most important parameters of the whole experiment are captured and recorded.**

plungaroni reviewed (changes requested):
Hi @rasool-thesis I saw your code and made some comments.
See if you can address some issues.

Maybe @skorpion17 can take a look in the meantime.

rasool-thesis committed:
feat: add per-CPU softirq processing delay reporting to user-space tool

- Extended the user-space netrace tool to report per-CPU processing delay and average delay for network softirqs.
- struct record updated to include arrays for latest delay, total delay, and delay count per CPU.
- stats_print() now outputs per-CPU load, last observed delay, and average delay (all in ms).
- Updated skeleton and header includes to match netrace-com BPF changes.
- Improved allocation and cleanup logic to handle new statistics fields.
- All changes fully commented for maintainability and team review.

skorpion17 commented on scripts/netrace-com:
Why did you load  a binary file?

rasool-thesis replied:
just to keep it, for using in pastrami

rasool-thesis replied:
I will remove it after completing my job.

rasool-thesis replied:
as before when we start nerace we use it for start. in start_netrace we use this with command /root/netrace | tee cpu_load_exp_id_${EXP_ID}.txt & to start , so I compile the file and load it to use instead of the netrace file as before.

rasool-thesis committed:
Create rtt_64byte_solveitteration_ipv4_fix.py

creating the IPV4 python file for testing 64B packets

rasool-thesis committed:
Add files via upload

The output of the test in SCV format

rasool-thesis committed:
create IPV6 python

script of the IPV6 for testing the 64B packets

rasool-thesis committed:
creating mac test or(L2)

creating the test script act in L2 with 64B packets

rasool-thesis committed:
creating script

create script in IPV4 and 1500B packets

rasool-thesis committed:
create script in L2 

create the script in L2 with 1500B packets.

rasool-thesis committed:
output 64B L2

output of the test script in 64B packets in L2

rasool-thesis committed:
output of IPV6 64B

output of IPV6 with 64B packets

rasool-thesis committed:
output of IPV4 1500B

output of test in IPV4 1500B

rasool-thesis committed:
output of L2 1500B

test output in 1500B L2

rasool-thesis committed:
the runner of tests

With this Python file, each test script can run easily through Trex

rasool-thesis committed:
Create 1

rasool-thesis committed:
output of 64B packets IPV4

rasool-thesis committed:
Delete latency/output/1

no need

rasool-thesis committed:
all the output of the scripts

rasool-thesis committed:
Delete latency/trex_iter_results_1500B_L2only.csv

move to output dir

rasool-thesis committed:
Delete latency/trex_iter_results_1500B_ipv4_fix.csv

move to output dir

rasool-thesis committed:
Delete latency/trex_iter_results_64B_L2only.csv

move to output dir

rasool-thesis committed:
Delete latency/trex_iter_results_64B_ipv4_fix.csv

move to output dir

rasool-thesis committed:
Delete latency/trex_iter_results_ipv6_udp.csv

move to output dir

rasool-thesis committed:
Update README.md

updating

rasool-thesis committed:
Update README.md

update

rasool-thesis committed:
Create new

rasool-thesis committed:
Add files via upload

rasool-thesis committed:
Delete images/new

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

update

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Delete scripts/TrexDriverCLI4.py

no need

rasool-thesis committed:
Delete scripts/TrexDriverCLI4_gpt_onethread.py

no need

rasool-thesis committed:
Delete scripts/config.py

no need

rasool-thesis committed:
Delete scripts/cpu_load_netrace.py

no need

rasool-thesis committed:
Delete scripts/cpu_load_netrace_gpt.py

no need

rasool-thesis committed:
Delete scripts/experiment-run-withconfig.py

no need

rasool-thesis committed:
Delete scripts/experiments_script_cpu_load_netrace-3.sh

no need

rasool-thesis committed:
Delete scripts/setup_tg.sh

no need

rasool-thesis committed:
Delete scripts/trex_installer.sh

no need

rasool-thesis committed:
Delete scripts/trex-3.04_run.sh

no need

rasool-thesis committed:
Delete scripts/start_netrace.sh

no need

rasool-thesis committed:
Delete scripts/start_netrace.py

no need

rasool-thesis committed:
Add files via upload

everything has been updated and the changes needed has done

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update rtt_1500byte_solveitteration_ipv4_fix.py

fixing the mac

rasool-thesis committed:
Update rtt_1500byte_solveitteration_maconly.py

fixing mac

rasool-thesis committed:
Update rtt_64byte_solveitteration_ipv4_fix.py

fixing the mac

rasool-thesis committed:
Update rtt_64byte_solveitteration_ipv6.py

fixing the mac

rasool-thesis committed:
Update rtt_64byte_solveitteration_maconly.py

fixing the mac

rasool-thesis committed:
Update rtt_1500byte_solveitteration_ipv4_fix.py

trex address fix

rasool-thesis committed:
Update rtt_1500byte_solveitteration_maconly.py

trex address fix

rasool-thesis committed:
Update rtt_64byte_solveitteration_ipv4_fix.py

trex address fix

rasool-thesis committed:
Update rtt_64byte_solveitteration_ipv6.py

trex address fix

rasool-thesis committed:
Update rtt_64byte_solveitteration_maconly.py

trex address fix

rasool-thesis committed:
Update run-latency.py

trex address fix

rasool-thesis committed:
Update TrexDriver.py

change absolute path for pcap file

rasool-thesis committed:
Update experiment-run-yaml.py

made str of the input value for CPU

rasool-thesis committed:
Create new

rasool-thesis committed:
Add files via upload

upload pcap file to avoid other requirement for path

rasool-thesis committed:
Delete scripts/pcap/new

no need

rasool-thesis committed:
Update trex_installer.sh

change the installation format

rasool-thesis committed:
Update README.md

make ssh in a good way

rasool-thesis committed:
Update README.md

make ssh fine description.

rasool-thesis committed:
Update README.md

clarify the outputs and requirements.

rasool-thesis committed:
Add files via upload

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update setup_tg.sh

rasool-thesis committed:
Create oneclick.py

creating fully automation.

rasool-thesis committed:
Update config.yml

adding port name for changing if needed

rasool-thesis committed:
Create rename_interface.py

rename interfaces if it was not match

rasool-thesis committed:
Create config_sut.yml

config interface name of sut

rasool-thesis committed:
Create rename_interface_sut.py

rename interfaces if needed for sut

Commit:
Fix: use RSA key + README update (ssh key location)

rasool-thesis committed:
Delete scripts/sshkeys/tg_to_sut

rasool-thesis committed:
Delete scripts/sshkeys/tg_to_sut.pub

rasool-thesis committed:
Update config.yml

solve hardcode

rasool-thesis committed:
Update experiment-run-yaml.py

make private key in non hardcode mode

Commit:
rebuild the latency

rasool-thesis committed:
Update sut_init.sh

update the setting for latency

Commit:
instalation for single latency

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Create README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Create README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update experiment-run-yaml.py

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md

rasool-thesis committed:
Update README.md
